corehq/apps/userreports/tasks.py

```diff
@@ -16,65 +16,90 @@
 from corehq.apps.userreports.sql import IndicatorSqlAdapter
 
 
-@task(queue='ucr_queue', ignore_result=True, acks_late=True)
-def rebuild_indicators(indicator_config_id):
-    def _is_static(config_id):
-        return config_id.startswith(StaticDataSourceConfiguration._datasource_id_prefix)
-
-    def _get_redis_key_for_config(config):
-        if _is_static(config._id):
-            rev = 'static'
-        else:
-            rev = config._rev
-        return 'ucr_queue-{}:{}'.format(config._id, rev)
-
-    is_static = _is_static(indicator_config_id)
-    if is_static:
-        config = StaticDataSourceConfiguration.by_id(indicator_config_id)
+def _is_static(config_id):
+    return config_id.startswith(StaticDataSourceConfiguration._datasource_id_prefix)
+
+
+def _get_config_by_id(indicator_config_id):
+    if _is_static(indicator_config_id):
+        return StaticDataSourceConfiguration.by_id(indicator_config_id)
     else:
-        config = DataSourceConfiguration.get(indicator_config_id)
+        return DataSourceConfiguration.get(indicator_config_id)
 
-    adapter = IndicatorSqlAdapter(config)
 
+def _get_redis_key_for_config(config):
+    if _is_static(config._id):
+        rev = 'static'
+    else:
+        rev = config._rev
+    return 'ucr_queue-{}:{}'.format(config._id, rev)
+
+
+def _build_indicators(indicator_config_id, relevant_ids):
+    config = _get_config_by_id(indicator_config_id)
+    adapter = IndicatorSqlAdapter(config)
     couchdb = _get_db(config.referenced_doc_type)
-    client = get_redis_client().client.get_client()
+    redis_client = get_redis_client().client.get_client()
```

Comment:
In reality these calls probably don't mean much (I have no idea about the `adapter` call), but you'd be repeating db hits (although probably cached?) every time a data source is rebuilt.

This would also probably be a lot of parameters to pass into a function as well. The javascript way would probably be to pass a function into `iter_docs` and use the variables defined outside the function, but not sure if that's pythonic